src/CLI/Encrypt_Decrypt/main.cpp

```diff
@@ -1,417 +1,15 @@
-#include "../../../src/lib/SoftHSM.h"
-//#include <corecrt_io.h>// Include the correct SoftHSM header
-#include <fcntl.h>
-#include <cstdlib>
-#include <iostream>
-#include <vector>
-#include <fstream>
-#include <filesystem>
-#include <cstring>
 
-typedef enum
-{
-	/** Operation was successful */
-	kStatus_SSS_Success = 0x5a5a5a5au,
-	/** Operation failed */
-	kStatus_SSS_Fail = 0x3c3c0000u,
-	/** Operation not performed because some of the passed parameters
-	 * were found inappropriate */
-	kStatus_SSS_InvalidArgument = 0x3c3c0001u,
-	// LCOV_EXCL_START
-	/** Where the underlying sub-system supports multi-threading,
-	 * Internal status to handle simultaneous access.
-	 *
-	 * This status is not expected to be returned to higher layers.
-	 * */
-	kStatus_SSS_ResourceBusy = 0x3c3c0002u,
-	// LCOV_EXCL_STOP
-} sss_status_t;
-
-namespace fs = std::filesystem;
-
-#define FILL_ATTR(attr, typ, val, len) {(attr).type=(typ); (attr).pValue=(val); (attr).ulValueLen=len;}
-#define S_IRUSR 0400
-#define S_IWUSR 0200
-
-#define MAGIC "35"
-#define GET_KEK_BT_INFO "105"
-#define ENCRYPT_DATA_KEY "107"
-#define DECRYPT_DATA_KEY "108"
-
-// Create an instance of SoftHSM
-SoftHSM* hsm = SoftHSM::i(); // Get the instance
-
-void intToBytes(int value, unsigned char* bytes) {
-	for (int i = 0; i < sizeof(int); i++) {
-		bytes[i] = (value >> (8 * i)) & 0xFF;
-	}
-}
-
-int gen_key(CK_SLOT_ID slot, CK_SESSION_HANDLE session, CK_OBJECT_HANDLE* hSecretKey)
-{
-	static int count = 0; //give the key id
-	count++;
-
-	CK_RV rv;
-	int key_length = 0;
-	do
-	{
-		std::cout << "enter the length of the key: 32/ 16/ 24 \n";
-		std::cin >> key_length;
-	} while (key_length != 32 && key_length != 16 && key_length != 24);
-	
-	//call NXP
-	uint8_t* arrRandom = new uint8_t[key_length]();
-	int status = GetRandom(arrRandom, key_length);
-	if (status != kStatus_SSS_Success)
-	{
-		std::cerr << "GetRandom failed";
-		exit(1);
-	}
+#pragma once
 
-	//call MSP
-	uint8_t kekId = Get - Kek - By - Info(MAGIC, GET_KEK_BT_INFO, data_lenght, (userId, kekInfo from the token), crc);
-	uint8_t * arrEncrypted = Encrypt - data - key(MAGIC, ENCRYPT_DATA_KEY, data_lenght, (userId, kekId, key_length, arrRandom), crc);
-
-	//cast to byte
-	unsigned char* countBytes;
-	unsigned char* kekIdBytes;
-	intToBytes(count, countBytes);
```

Comment:
std::copy that what you want? if you want to change order of big/little endian use htonl etc.